pkgs/by-name/op/open-adventure/package.nix

```
@@ -0,0 +1,84 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitLab,
+  python3Packages,
+  pkg-config,
+  libedit,
+  cppcheck,
+  coreutils,
+  asciidoctor,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "open-adventure";
+  version = "1.20";
+  src = fetchFromGitLab {
+    owner = "esr";
+    repo = "open-adventure";
+    tag = finalAttrs.version;
+    hash = "sha256-xbsMz99CNLhpM6BSJVcRzxPB6tUYfPy/3Z+8BKt8b1E=";
+  };
+
+  nativeBuildInputs = [
+    python3Packages.python
+    python3Packages.pyyaml
+    pkg-config
+    asciidoctor
+  ];
+
+  buildInputs = [ libedit ];
+
+  doCheck = true;
+
+  nativeCheckInputs = [
+    python3Packages.pylint
+    cppcheck
+  ];
+
+  postPatch = ''
+    patchShebangs --build make_dungeon.py
+
+    # Fix the lack of the = symbol in the cli of cppcheck
+    substituteInPlace Makefile --replace-fail "--template " "--template="
+
+    # Replace /bin/echo with the actual path in the nix store
+    substituteInPlace tests/tapview --replace-fail "/bin/echo" ${lib.getExe' coreutils "echo"}
+  '';
+
+  postBuild = ''
+    make advent.6
+  '';
+
+  preInstall = ''
+    mkdir -vp "$out/bin" "$out/share/man/man6" "$out/share/applications/" "$out/share/icons/hicolor/scalable/apps"
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    cp ./advent $out/bin
+    runHook postInstall
+  '';
+
+  postInstall = ''
+    cp ./advent.6 $out/share/man/man6
+    cp ./advent.desktop $out/share/applications
+    cp ./advent.svg $out/share/icons/hicolor/scalable/apps
+  '';
```

```suggestion
  installPhase = ''
    runHook preInstall
    
    mkdir -vp "$out/bin" "$out/share/man/man6" "$out/share/applications/" "$out/share/icons/hicolor/scalable/apps"
    cp ./advent $out/bin
    cp ./advent.6 $out/share/man/man6
    cp ./advent.desktop $out/share/applications
    cp ./advent.svg $out/share/icons/hicolor/scalable/apps
    
    runHook postInstall
  '';
```

This should also use the `install` command whenever possible. I don't know the package filesystem layout to suggest an implementation.